content/articles/aes-rsa-encryption/index.md

```diff
@@ -0,0 +1,138 @@
+
+### Getting started with RSA encryption algorithm
+
+RSA was named after its first describers Rivest, Shamir, and Adleman. Rivest–Shamir–Adleman is an asymmetric encryption algorithm used by modern computers in data encryption and decryption. RSA  uses two keys that are the public key that is used for encryption and the private key used for data decryption hence is an asymmetric algorithm. The encryption key is always made public hence is accessed by anyone while the decryption key is kept private.
+
+This algorithm is based on the fact that finding factors of [composite numbers](https://en.wikipedia.org/wiki/Composite_number) is a difficult task.
+
+Below is a simple diagrammatic representation flow of how RSA encrypts data:
+![RSA structure](](/engineering-education/aes-rsa-encryption/RSA.png)
+
+### RSA algorithm functionality
+
+Below is the step by step instructions of how to achieve encryption using the RSA algorithm:
+
+ ### Step 1: Generation of the encryption and decryption keys
+
+ To generate the keys,
+ - Pick any two large prime numbers and keep them secret.
+
+     Example lets pick  prime numbers x and y
+- Evaluate 
+              
+     z = xy where z is the modulus of the two keys being generated
+
+- Evaluate the [totient](https://simple.wikipedia.org/wiki/Euler%27s_totient_function)
+
+    φ(z)=(x-1)(y-1)
+
+- Pick any integer e in which 1 < e <  φ(z) and e is the public key exponent
+
+- Evaluate  k  and keep it as the private key exponent as follows:
+
+    ke is [congruent](https://en.wikipedia.org/wiki/Modular_arithmetic) 1 in mod φ(z) 
+
+   ke ≡ 1 (mod φ(z))
+
+
+
+
+The public key is made of the modulus z and the exponent e while the private key is made of x and y and the private exponent k.
+The public key is made available to everyone but the private key should only be accessed by the recipient of the message.
+
+### Step 2: Encrypting the message using the generated encryption key
+
+ - Send the public key (z,w) generated above to the person who wants to send a message and remain with the private key.
+ - The message sender turns the message into a number "G". The number G should be smaller than "z". This is done using an agreed-upon protocol commonly referred to as [Padding scheme](https://simple.wikipedia.org/wiki/RSA_algorithm#Padding_schemes)
+ - The sender then computes the ciphertext of the message using the expression, c = G^e  mod z where 'c' represents the ciphertext.
+
+The receiver can decrypt the message using the expression G = c^d  mod n according to the [chinese remainder theorem](https://simple.wikipedia.org/wiki/Chinese_remainder_theorem).
+
+### Conclusion
+
+Encryption is key in matters of information security, and it is vital to learn the different algorithms used in data encryption other than AES and RSA to broaden your horizons as a person in the computing world where cybercrime is increasing by the hour.
```

Comment:
Avoid the phrases "In conclusion" and "In summary". Phrases like these usually come across as cliche'. Also, your conclusion statement is too long. Consider breaking it into shorter sentences.